implementations/c/ockam/transport/posix_socket/socket_tcp.c

```diff
@@ -255,7 +255,7 @@ ockam_error_t socket_tcp_read(void* ctx, uint8_t* buffer, size_t buffer_size, si
     bytes_read        = recv(p_socket->socket_fd, &recv_len, sizeof(uint16_t), 0);
     // Must convert from network order to native endianness
     p_transmission->transmit_length = ntohs(recv_len);
-    if (-1 == bytes_read) {
+    if (sizeof(uint16_t) != bytes_read) {
```

Comment:
Does casting (-1) to an unsigned number saturate to 0 or uint16_t.MAX? If so it saturates high, this could be a problem.

Reply:
Actually this should go to #295 . I planned that we could merge #295 first, and then transport changes will disappear from here